src/main/java/org/jboss/ejb/protocol/remote/RemoteServer.java

```diff
@@ -539,7 +546,7 @@ public void writeResponse(final Object response) {
             }
         }
 
-        public void writeException(final Exception response) {
+        public void writeThrowable(final Throwable response) {
```

Comment:
Current org.jboss.as.ejb3.remote.protocol.versionone.SessionOpenRequestHandler catches Throwable during session id generation so I assumed that some Errors may be thrown during session id generation.